src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -145,7 +146,7 @@ public SQLiteStore(String directory, Manager manager, StoreDelegate delegate) {
         }
         this.path = new File(directory, kDBFilename).getPath();
         this.manager = manager;
-        this.storageEngine = null;
+        this.storageEngine = createStorageEngine();
```

Comment:
It's required because the storage engine now is needed for doing key derivation which will happen before open() method is called.

https://github.com/couchbase/couchbase-lite-java-core/blob/feature/issue_805_key_derivation/src/main/java/com/couchbase/lite/Database.java#L1080-L1088